packages/fxa-auth-server/lib/oauth/db/mysql/index.js

```diff
@@ -187,9 +190,16 @@ class MysqlStore extends MysqlOAuthShared {
     }
   }
 
-  getLock(lockName, timeout = 3) {
-    // returns `acquired: 1` on success
-    return this._readOne(QUERY_GET_LOCK, [lockName, timeout]);
+  async _withLock(cb, lockName, timeout = 3) {
```

Comment:
MySQL's `RELEASE_LOCK` needs to be called in the same session as the `GET_LOCK` call that obtained the lock.